google-beta/resource_storage_bucket_access_control.go

```diff
@@ -0,0 +1,375 @@
+// ----------------------------------------------------------------------------
+//
+//     ***     AUTO GENERATED CODE    ***    AUTO GENERATED CODE     ***
+//
+// ----------------------------------------------------------------------------
+//
+//     This file is automatically generated by Magic Modules and manual
+//     changes will be clobbered when the file is regenerated.
+//
+//     Please read more about how to change this file in
+//     .github/CONTRIBUTING.md.
+//
+// ----------------------------------------------------------------------------
+
+package google
+
+import (
+	"fmt"
+	"log"
+	"reflect"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceStorageBucketAccessControl() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceStorageBucketAccessControlCreate,
+		Read:   resourceStorageBucketAccessControlRead,
+		Update: resourceStorageBucketAccessControlUpdate,
+		Delete: resourceStorageBucketAccessControlDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceStorageBucketAccessControlImport,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(4 * time.Minute),
+			Update: schema.DefaultTimeout(4 * time.Minute),
+			Delete: schema.DefaultTimeout(4 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"bucket": {
+				Type:             schema.TypeString,
+				Required:         true,
+				DiffSuppressFunc: compareSelfLinkOrResourceName,
+			},
+			"entity": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"entity_id": {
```

Comment:
It's my understanding that this field is read-only. Did you find that wasn't the case?

Reply:
This is primarily based on https://cloud.google.com/storage/docs/json_api/v1/bucketAccessControls/insert, which says that only `entity` and `role` are configurable.